test/integration/good/code-gen/cpp.expected

```diff
@@ -12030,6 +12030,887 @@ stan::model::model_base& new_model(
+    
+    try {
+      
+      pos__ = 1;
+      context__.validate_dims("data initialization","n","int",
+          context__.to_vec());
+      
+      current_statement__ = 19;
+      pos__ = 1;
+      current_statement__ = 19;
+      n = context__.vals_i("n")[(pos__ - 1)];
+      context__.validate_dims("data initialization","x","double",
+          context__.to_vec());
+      
+      current_statement__ = 20;
+      pos__ = 1;
+      current_statement__ = 20;
+      x = context__.vals_r("x")[(pos__ - 1)];
+    } catch (const std::exception& e) {
+      stan::lang::rethrow_located(e, locations_array__[current_statement__]);
+      // Next line prevents compiler griping about no return
+      throw std::runtime_error("*** IF YOU SEE THIS, PLEASE REPORT A BUG ***"); 
+    }
+    num_params_r__ = 0U;
+    
+    try {
+      num_params_r__ += 1;
+      num_params_r__ += 1;
+    } catch (const std::exception& e) {
+      stan::lang::rethrow_located(e, locations_array__[current_statement__]);
+      // Next line prevents compiler griping about no return
+      throw std::runtime_error("*** IF YOU SEE THIS, PLEASE REPORT A BUG ***"); 
+    }
+  }
+  template <bool propto__, bool jacobian__, typename T__>
+  T__ log_prob(std::vector<T__>& params_r__, std::vector<int>& params_i__,
+               std::ostream* pstream__ = 0) const {
+    typedef T__ local_scalar_t__;
+    T__ lp__(0.0);
+    stan::math::accumulator<T__> lp_accum__;
+    static const char* function__ = "truncate_model_namespace::log_prob";
+(void) function__;  // suppress unused var warning
+
+    stan::io::reader<local_scalar_t__> in__(params_r__, params_i__);
+    
+    try {
+      local_scalar_t__ m;
+      
+      current_statement__ = 1;
+      m = in__.scalar();
+      local_scalar_t__ y;
+      
+      current_statement__ = 2;
+      y = in__.scalar();
+      current_statement__ = 2;
+      if (jacobian__) {
+        current_statement__ = 2;
+        y = stan::math::lb_constrain(y, 0, lp__);
+      } else {
+        current_statement__ = 2;
+        y = stan::math::lb_constrain(y, 0);
+      }
+      {
+        current_statement__ = 3;
+        lp_accum__.add(normal_log<propto__>(x, m, 1));
+        current_statement__ = 4;
+        if (logical_lt(x, 0.0)) {
+          current_statement__ = 4;
+          lp_accum__.add(stan::math::negative_infinity());
+        } else {
+          current_statement__ = 4;
+          lp_accum__.add(-normal_ccdf_log(0.0, m, 1));
+        }
+        current_statement__ = 5;
+        lp_accum__.add(normal_log<propto__>(x, m, 1));
+        current_statement__ = 6;
+        if (logical_gt(x, 10.0)) {
+          current_statement__ = 6;
+          lp_accum__.add(stan::math::negative_infinity());
+        } else {
+          current_statement__ = 6;
+          lp_accum__.add(-normal_cdf_log(10.0, m, 1));
+        }
+        current_statement__ = 7;
+        lp_accum__.add(normal_log<propto__>(x, m, 1));
+        current_statement__ = 9;
```

Comment:
These statement numbers refer to the locations array on line 12515. By my count these are
* 8 -> (in 'truncate.stan', line 13, column 26 to column 30)
* 9 -> (in 'truncate.stan', line 13, column 22 to column 25)
* 10 -> (in 'truncate.stan', line 13, column 4 to column 32)

so looks like they are 8=lower expression, 9=upper expression, and 10=full statement. The regression from 9 to 8 happens because the upper bound is checked before the lower bound. To be honest I didn't think about the locations, just grabbed the nearest metadata around. So now exceptions thrown by cdfs are associated with the lower bound regardless of which cdf call failed.